File: TrackIt.Model/Models/Goal.cs
namespace TrackIt.Model.Models
{
    public class Goal
    {
        public int GoalId { get; set; }

        public string? Name { get; set; }

        public string? Description { get; set; }

        public double? TargetProtein { get; set; }

        public int? TargetCalories { get; set; }
    }
}

File: UI/trackit_admin/lib/models/Goal/goal.dart
import 'package:json_annotation/json_annotation.dart';
part 'goal.g.dart';

@JsonSerializable()
class Goal {
  int? goalId;
  String? name;
  String? description;
  int? targetProtein;
  double? targetCalories;
  String? image;

  Goal(this.goalId, this.name, this.description, this.targetProtein,
      this.targetCalories);

  factory Goal.fromJson(Map<String, dynamic> json) => _$GoalFromJson(json);

  Map<String, dynamic> toJson() => _$GoalToJson(this);
}
